impeller/playground/imgui/imgui_impl_impeller.cc

```diff
@@ -150,14 +152,43 @@ void ImGui_ImplImpeller_RenderDrawData(ImDrawData* draw_data,
        draw_list_i++) {
     const ImDrawList* cmd_list = draw_data->CmdLists[draw_list_i];
 
+    // Convert ImGui's per-vertex data (`ImDrawVert`) into the per-vertex data
+    // required by the shader (`VS::PerVectexData`). The only difference is that
+    // `ImDrawVert` uses an `int` for the color and the impeller shader uses 4
+    // floats (since GLSL ES 1 doesn't support integer vertex attributes).
+    //
+    // Once the GLES 2 renderer lands, impellerc could be extended to map the 4
+    // color bytes to vec4s in the shader, i.e.:
+    //     glVertexAttribPointer(2, 4, GL_UNSIGNED_BYTE, false, 4, nullptr);
+    // ...as opposed to:
+    //     glVertexAttribPointer(2, 4, GL_FLOAT, false, 16, nullptr);
+    //
+    // impellerc could e.g. detect some kind of special naming convention hint
+    // from the attribute name to support this case:
+    //     in vec4 vertex_color__byte;
+    //
+    // Such a feature would remove the need to double-copy/convert this data.
+
+    std::vector<VS::PerVertexData> vtx_data;
+    vtx_data.reserve(cmd_list->VtxBuffer.size());
+    for (const auto& v : cmd_list->VtxBuffer) {
+      const impeller::Scalar kScale = 1.0f / 255.0f;
+      impeller::Vector4 color(((v.col >> 0) & 0xFF) * kScale,   //
```

Comment:
Nit: `ColorConvertU32ToFloat4` instead. I bet it does exactly the same thing but I see defines in there to check to see if the format is BGRA instead. Good to not have this be tied to RGBA here.

Reply:
Done.